### PR TITLE
Start a release process documentation.

### PR DESCRIPTION
**Pull request by Zimmi48**
These notes were largely inspired by [the release process documentation of NixOS](https://nixos.org/nixos/manual/index.html#release-process).

It is only a beginning, a lot of things are missing. When this is finished this should allow to check one point of #5976.
I am creating this now, so that we don't forget again to put the alpha tag when branching.

### CODE REVIEW

**Comment by psteckler** [comment]
Should there be anything here about the XML protocol version, or notifying Proof General maintainers (some code examines the Coq version)?

**Comment by Zimmi48** [comment]
How does the XML protocol version work? Is it something that should be increased at each release? If yes, it would make sense to include it here. If not, then this is unrelated I guess.

**Comment by ejgallego** [comment]
The idea is that the IDE protocols will be basically tied 1:1 to the document manager API, so IMHO it should be handled outside the release process as you point out.

**Comment by Zimmi48** [comment]
As promised, here is the release documentation. These steps only represent a proposed process (mostly corresponding to what was done for the two previous releases 8.7 and 8.8).

File: dev/doc/release-process.md
Comment (Zimmi48) [inline dev/doc/release-process.md]: This is only a proposal to have a good way to track the release process. I think it is a good idea but it was never experimented by @maximedenes and I (because this checklist wasn't ready yet).
@silene Tell me if you think you will use this way of doing. If not, there is no point in keeping this step in the doc.

**Comment by SkySkimmer** [comment]
New file -> new codeowner entry?

File: dev/doc/release-process.md
Comment (SkySkimmer) [inline dev/doc/release-process.md]: IIRC @cpitclaudel was missed by git for the sphinx stuff, maybe a note that the command yields an approximation would be good?

**Comment by Zimmi48** [comment]
> New file -> new codeowner entry?

It would certainly be a good rule many times but in this case, the new file is in the scope of `dev/doc` of which I'm already a code owner.

File: dev/doc/release-process.md
Comment (Zimmi48) [reply dev/doc/release-process.md]: Maybe, but I'd rather make sure that people properly set the "author" of their commits in the future.

**Comment by Zimmi48** [comment]
I consider this document ready on my side and I am waiting for feedback. The only related TO-DOs are to automate the release more (cf. https://github.com/coq/coq/projects/23).

File: dev/doc/release-process.md
Comment (silene) [reply dev/doc/release-process.md]: It will depend whether the process happens to be linear or not. If it is strictly linear, I presumably won't bother creating a new issue. If I end up having to track several items in parallel, then yours is a good idea and I will copy this document.

**Comment by silene** [comment]
Overall, the document is fine to me. Some specific points are not that clear to me, but experience will tell.

**Review by silene — APPROVED** [review]

**Review by maximedenes — APPROVED** [review]
Looks like a good start, thanks!

**Pull request merged**
